doc/nrf/releases_and_maturity/migration/migration_guide_2.4_to_2.5.rst

```diff
@@ -0,0 +1,28 @@
+:orphan:
+
+.. _migration_2.5:
+
+Migration guide for |NCS| v2.5.0 (Working draft)
+################################################
+
+This document describes the changes required or recommended when migrating your application from |NCS| v2.4.0 to |NCS| v2.5.0.
```

Comment:
Perhaps we should explain what is the intent behind required and recommended changes? For required it seems to be pretty clear: one needs these changes to make the code build or perform in the same way as in previous release.

But what is the reasoning for the recommended changes?

Reply:
Added this.